### DISCUSSION TITLE
Allow custom order of categories in category fields

### DISCUSSION THREAD

**Discussion by aloco** [craftcms/cms#10560] [Ideas] [answered]
### Description

When using a field of type category I would like to order the selected categories on the category field the same way as for entries. I would expect to change the order of categories within the same level, defaulting to the order of the category type which can be set on the categories page of the CP. 

### Steps to reproduce

1. Create a category field
2. Select some categories 
3. Try to order the categories via drag and drop, like you can do with entries

### Additional info

- Craft version: Craft Pro 3.7.30.1
- PHP version: 7.4.16
- Database driver & version: MySQL 5.7.36
- Plugins & versions:

**Comment by brandonkelly** [comment]
This is currently working as intended, so converted to a discussion.

**Reply by aloco** [reply]
Hi Brandon, 

do you have any plans to make this possible in the future? If I understood correct, you mean it is not possible right now.

Thanks!

**Reply by brandonkelly** [reply]
We are considering making it configurable, yes.

**Reply by MattAppleton** [reply]
We were hoping this might be possible too — working on a new project today, where allowing the client to order categories per-entry would be a lovely feature :)

**Reply by alexanderloewe** [reply]
I would love to see this option included. 
However, I would also like to have the option to prevent free sorting.

Is it also planned that only when selecting level 2 or 3 categories, the parents can not be selected, like with an entry field?

**Reply by iparr** [reply]
I keep hitting situations on very different projects with _very_ different clients where easily giving priority to categories would be very nice, so giving this a 👍

**Reply by splendidrob** [reply]
Come up against this issue on every project, all clients ask why they cant re-order the categories they have selected...

**Reply by garycrane** [reply]
+1 for this. Guessing this will go away with 'entries everywhere' in Craft 5?

**Comment by brandonkelly** [comment]
This was previously requested by #8748, so marking this as a duplicate.

**Comment by brandonkelly** [comment, marked as answer]
This has been resolved for Craft 4.4, which adds a new “Maintain hierarchy” setting to Categories and Entries fields, giving them control over whether the specific order/hierarchy of the category/entry structure should be enforced by the field. (#11088)

**Reply by garycrane** [reply]
Legend. Thanks so much Brandon.

**Reply by brandonkelly** [reply]
Craft 4.4 is out with that change!

**Reply by alexanderloewe** [reply]
Like ;)

**Reply by MattAppleton** [reply]
Great! Thank you!

**Reply by grtdb** [reply]
I have unchecked the _Maintain hierarchy_  setting on one of our Category fields, but I cannot see how to reorder the selected categories; I have tried dragging them and nothing happens and I can't see any other controls.  Am I missing something?

**Reply by iparr** [reply]
Same here @grtdb - this doesn't match what I was expecting I'm afraid:

- I can't re-order categories with drag/drop.
- If I select a 'leaf' at the end of a 'tree' I expected that and only that to be selected. It is, but then when I hit 'save' its parents re-appear (albeit without any suggestion of hierarchy).
- If I removed a 'branch' between the 'leaf' and the 'trunk' (god, I've got to get better analogies for this), then hit the remove cross, then it will go but upon save it re-appears. I think i know why this is happening but I can imagine it is potentially very confusing for non-technical editors.

Very sorry! I personally don't understand the use-case for this behaviour as this wouldn't allow me to do anything different with this field in the front-end either, right?

**Reply by brandonkelly** [reply]
@grtdb @iparr Sorry, that was a bug. Fixed for the next release (9dc65118eb783b282b320450baa16bae93e5b6b4)

**Reply by brandonkelly** [reply]
Craft 4.4.6 is out now with that fix 🎉

**Reply by iparr** [reply]
Great. I can now re-order categories with drag/drop. However, but my other points still remain. Upon save, the position I have dragged them into is forgotten.

**Reply by brandonkelly** [reply]
Sorry… not sure how I missed that. 4.4.6.1 is out now with a complete fix.
